src/screenComponents/hackingDialog.cpp

```diff
@@ -128,115 +133,41 @@ bool GuiHackingDialog::onMouseDown(sf::Vector2f position)
     return true;
 }
 
-void GuiHackingDialog::resetMinigame()
+void GuiHackingDialog::onMiniGameComplete(bool success)
 {
-    for(int x=0; x<hacking_field_size; x++)
-    {
-        for(int y=0; y<hacking_field_size; y++)
-        {
-            FieldItem& item = field_item[x][y];
-            item.button->setText("");
-            item.button->setValue(false);
-            item.button->enable();
-            item.bomb = false;
-        }
-    }
-    for(int n=0; n<bomb_count; n++)
-    {
-        int x = irandom(0, hacking_field_size - 1);
-        int y = irandom(0, hacking_field_size - 1);
-        
-        if (field_item[x][y].bomb)
-        {
-            n--;
-            continue;
-        }
-        field_item[x][y].bomb = true;
-    }
-    error_count = 0;
-    correct_count = 0;
-    
-    progress_bar->setValue(0.0f);
-    status_label->setText("Hacking in progress: 0%");
-    reset_button->enable();
+    reset_time = engine->getElapsedTime() + auto_reset_time;
+    game->disable();
+    last_game_success = success;
+    status_label->setText(success ? "Hacking SUCCESS!" : "Hacking FAILURE!");
 }
 
-void GuiHackingDialog::disableMinigame()
-{
-    status_label->setText("Select hacking target...");
-    
-    for(int x=0; x<hacking_field_size; x++)
-    {
-        for(int y=0; y<hacking_field_size; y++)
-        {
-            FieldItem& item = field_item[x][y];
-            item.button->setText("");
-            item.button->setValue(false);
-            item.button->disable();
-        }
+void GuiHackingDialog::getNewGame() {
+    delete game;
+    int difficulty = 2;
+    EHackingGames games = HG_All;
+    if (gameGlobalInfo) {
+      difficulty = gameGlobalInfo->hacking_difficulty + 1;
```

Comment:
Magic + 1?

Reply:
Yeah. Problem is: the selectors on the Server creation screen are 0-based, as far as I can tell, and a difficulty of 0 doesn't make much sense for the way the games are defined (Both have an offset, but LightsOut comes out as a 1x1 grid with difficulty 0, which is... sort of trivial, and MineSweeper gets 4x4, which isn't much better). I'm not altogether happy with the magic difficulty generation for each separate minigame either, so I'll probably move that, and the associated magic numbers to the individual game implementations.

Reply:
Moved the adjustment to the definition of the grid_size variables. That way, the original difficulty is still accessible if it's ever needed, and if someone wants, it's easy to make bomb counts different from grid size or similar.